### PR TITLE
#17 implement login page

### PR DESCRIPTION
**Pull request by ConnyPettersson**
- Add a Tailwind-styled login page integrating GitHub OAuth2
- Created a dedicated login page component styled with Tailwind CSS
- Integrated GitHub OAuth2 login flow to enable user authentication
- Improved user experience with clean layout, center-aligned content, and a clear call-to-action button
- Ensured environment variables for OAuth2 credentials are read from .env

## Summary by CodeRabbit

- **New Features**
	- Integrated Tailwind CSS framework for enhanced styling capabilities.

- **Bug Fixes**
	- Minor adjustment to the export statement of the App component for consistency.

### CODE REVIEW

File: frontend/src/LoginPage.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+function LoginPage() {
+    return (
+        <div className="min-h-screen flex flex-col items-center justify-center bg-gray-50">
+            <h1 className="text-3xl font-bold mb-6 text-gray-800">Welcome to System2024!</h1>
+            <p className="mb-8 text-gray-600">Sign in with your GitHub account to continue</p>
+            <a href="/oauth2/authorization/github">
+                <button className="bg-blue-600 text-white py-2 px-4 rounded hover:bg-blue-700 transition">Sign in with
+                    GitHub
+                </button>
+            </a>
```
[the remark this answers is not in the document]
Comment (MikaelaMathisson) [reply frontend/src/LoginPage.tsx:11]: I would look into maybe making these changes to improve accessibility 👍

**Review by oskber — DISMISSED** [review]
Looks good to me 👍 
There are some conflicts that needs to be solved before merge.

**Review by MikaelaMathisson — DISMISSED** [review]
I dont see any issues with this code. I think it can be merged after conflicts are resolved 💯

**Review by MikaelaMathisson — APPROVED** [review]
Approving!

**Review by oskber — DISMISSED** [review]
approving fix

**Review by MikaelaMathisson — DISMISSED** [review]
There was a conflict that i solved. Approving again

**Review by MatsJoh50 — DISMISSED** [review]
Great start for a login page.

I would like to see the LoginPage.tsx be moved to the /pages folder
and with that remove the component from App.tsx.

also add the component to main.tsx for routing. example to follow:

`  {
        path: '/auth/login',
        element: <LoginPage />,
    }`

**Review by MikaelaMathisson — DISMISSED** [review]
Nice work with fixing the provided suggestions from Mats 👍

**Comment by ConnyPettersson** [comment]
> Great start for a login page.
> 
> I would like to see the LoginPage.tsx be moved to the /pages folder and with that remove the component from App.tsx.
> 
> also add the component to main.tsx for routing. example to follow:
> 
> ` { path: '/auth/login', element: <LoginPage />, }`

Everything has been taken care of

**Review by MatsJoh50 — DISMISSED** [review]
Good job implementing suggestions.

**Review by oskber — DISMISSED** [review]
Approving fixes

**Review by MikaelaMathisson — APPROVED** [review]
Approving fixes!

**Review by MatsJoh50 — APPROVED** [review]
Seems to be correct now !

**Review by Weeezzt — COMMENTED** [review]
Looks good! Maybe you could add a if(error) and if(isLoading)  and return something like a simple text that lets you know whats happening?

**Pull request closed without merging**